src/main/java/it/pagopa/ecommerce/payment/methods/application/PaymentMethodService.java

```diff
@@ -270,26 +269,36 @@ public Mono<CreateSessionResponseDto> createSessionForPaymentMethod(String id) {
                     URI resultUrl = returnUrlBasePath.resolve(sessionUrlConfig.outcomeSuffix());
                     URI merchantUrl = returnUrlBasePath;
                     URI cancelUrl = returnUrlBasePath.resolve(sessionUrlConfig.cancelSuffix());
-                    String orderId = UUID.randomUUID().toString().replace("-", "").substring(0, 15);
                     String customerId = UUID.randomUUID().toString().replace("-", "").substring(0, 15);
+                    URI notificationUrl = UriComponentsBuilder
+                            .fromHttpUrl(sessionUrlConfig.notificationUrl())
+                            .build(
+                                    Map.of(
+                                            "orderId",
+                                            orderId,
+                                            "paymentMethodId",
+                                            id
+                                    )
+                            );
 
                     return npgClient.buildForm(
-                            correlationId,
-                            returnUrlBasePath,
-                            resultUrl,
-                            merchantUrl,
-                            cancelUrl,
-                            orderId,
-                            customerId,
-                            paymentMethod,
-                            npgDefaultApiKey
+                            correlationId, // correlationId
+                            merchantUrl, // merchantUrl
+                            resultUrl, // resultUrl
+                            notificationUrl, // notificationUrl
+                            cancelUrl, // cancelUrl
+                            orderId, // orderId
+                            customerId, // customerId
+                            paymentMethod, // paymentMethod
+                            npgDefaultApiKey // defaultApiKey
                     ).map(form -> Tuples.of(form, sessionPaymentMethod));
                 }).map(data -> {
                     FieldsDto fields = data.getT1();
 
                     npgSessionsTemplateWrapper
                             .save(
                                     new NpgSessionDocument(
+                                            orderId,
```

Comment:
Update also NpgSessionTemplateWrapper method to retrieve key from entity to use orderId [here](https://github.com/pagopa/pagopa-ecommerce-payment-methods-service/blob/a8749b9f0fc7fa02d8b90a50573f11a99cf32476/src/main/java/it/pagopa/ecommerce/payment/methods/infrastructure/NpgSessionsTemplateWrapper.java#L26)

Reply:
Fixed in  655043b